src/implementations/transaction/transaction.ts

```diff
@@ -0,0 +1,26 @@
+import signale from "signale";
+import { TransactionCancelCommand } from "../../commands/transaction/transaction-command.type";
+import { getWallet } from "../utils/wallet";
+import { BigNumber } from "ethers";
+
+export const cancelTransaction = async ({
```

Comment:
is it possible to write a comment block above this explaining how cancellation actually works?
from the chat messages ive read so far, as long as this new transaction with the same nonce happens at a higher priority (higher likely chance to be picked up by a miner, before the pending one that has not been picked up by a miner) and been processed, then the pending transaction will be cancelled (dropped by the miner) as the miner will drop any transactions with the nonce already processed.

btw shouldnt we hardcode the gas fee to a high number so cancellation will most likely to take priority?

is that right? @rjchow

Reply:
the problem with hardcoding is that gas fees are dynamic and may change beyond what you expect, but i'm ok with a default that's maybe max(2-5x of current estimated gas prices, 1.1* > this transaction's price), and let the user add a flag if he wants to overwrite

might find this article useful https://www.blocknative.com/blog/canceled-transactions

Reply:
hmm i see, i'll take a look at the link

Reply:
For the gas fee, I have tried 1.1 but they return an underflow error. So I have increased the cancellation transaction gas price 2x the current gas price.